pkg/cloud/instance.go

```diff
@@ -231,13 +266,49 @@ func (c *client) DestroyVMInstance(csMachine *infrav1.CloudStackMachine) error {
 
 	// Attempt deletion regardless of machine state.
 	p := c.cs.VirtualMachine.NewDestroyVirtualMachineParams(*csMachine.Spec.InstanceID)
+	volIDs, err := c.listVMInstanceVolumeIDs(*csMachine.Spec.InstanceID)
+	if err != nil {
+		return err
+	}
 	p.SetExpunge(true)
+	setArrayIfNotEmpty(volIDs, p.SetVolumeids)
 	if _, err := c.csAsync.VirtualMachine.DestroyVirtualMachine(p); err != nil &&
 		strings.Contains(strings.ToLower(err.Error()), "unable to find uuid for id") {
 		// VM doesn't exist. Success...
 		return nil
 	} else if err != nil {
 		return err
 	}
+
+	if err := c.ResolveVMInstanceDetails(csMachine); err == nil && (csMachine.Status.InstanceState == "Expunging" ||
```

Comment:
I'm pretty sure these changes are already in main? I think if you click edit and then change base, then don't really change the base it will refresh the diff?

Reply:
Cool, either way this branch is going to need some merging with main after the CRD branch goes through. Why don't we take another look then.